### PR TITLE
Database::get_entities_by_name [11287]

### PR DESCRIPTION
**Pull request by EduPonz**

### CODE REVIEW

File: src/cpp/database/database.hpp [hunk 1]
```diff
@@ -193,6 +193,35 @@ class Database
      */
     EntityId generate_entity_id() noexcept;
 
+    /**
+     * Get all entities of a given EntityKind that match with the requested name
+     *
+     * @param entity_kind The EntityKind of the fetched entities
+     * @param name The name of the entities for which to search
+     * @throws eprosima::statistics_backend::BadParameter if entity_kind is EntityKind::INVALID
+     *         eprosima::statistics_backend::Unsupported if entity_kind is not supported
```
Comment (IkerLuengo) [inline src/cpp/database/database.hpp:202]: I'd use `BadParameter` in both cases

File: src/cpp/database/database.hpp [hunk 2]
```diff
@@ -193,6 +193,35 @@ class Database
...
+    /**
+     * Get all entities of a given EntityKind that match with the requested GUID
+     *
+     * If the given EntityKind does not contain a GUID,
+     * BadParameter is thrown.
+     *
+     * @param entity_kind The EntityKind of the fetched entities
+     * @param guid The GUID of the entities to search for
+     * @return A vector of pairs, where the first field is the EntityId of the Domain of the matching entities,
+     *         and the second is the EntityId of the matching entities
```
Comment (IkerLuengo) [inline src/cpp/database/database.hpp:219]: I'd use `BadParameter` in both cases

File: test/unittest/Database/DatabaseTests.cpp [hunk 3]
```diff
@@ -2100,6 +2111,143 @@ TEST_F(database_tests, get_entity_no_existing)
     ASSERT_THROW(db.get_entity(EntityId()), BadParameter);
 }
 
+TEST_F(database_tests, get_entities_by_name_host)
+{
+    auto hosts = db.get_entities_by_name(EntityKind::HOST, host_name);
+    EXPECT_EQ(hosts.size(), 1);
+    EXPECT_FALSE(hosts[0].first.is_valid());
+    EXPECT_EQ(hosts[0].second, host_id);
+}
```
Comment (IkerLuengo) [inline test/unittest/Database/DatabaseTests.cpp:2120]: I'm missing tests where there is more than one entity, checking that we do not return just the first one we found.

File: src/cpp/database/database.hpp [hunk 4]
```diff
@@ -193,6 +193,35 @@ class Database
...
 
+    /**
+     * Get all entities of a given EntityKind that match with the requested name
+     *
+     * @param entity_kind The EntityKind of the fetched entities
+     * @param name The name of the entities for which to search
+     * @throws eprosima::statistics_backend::BadParameter if entity_kind is EntityKind::INVALID
+     *         eprosima::statistics_backend::Unsupported if entity_kind is not supported
+     * @return A vector of pairs, where the first field is the EntityId of the Domain of the matching entities,
+     *         and the second is the EntityId of the matching entities
```
Comment (IkerLuengo) [inline src/cpp/database/database.hpp:204]: Add a clarification like:
```
For physical entities (Host, User, Process, Locator) the returned Domain EntityId has no meaning, as these entities do not belong to a Domain
```

File: src/cpp/database/database.hpp (hunk 2 above)
Comment (IkerLuengo) [inline src/cpp/database/database.hpp:219]: Add a clarification like:
```
For physical entities (Host, User, Process, Locator) the returned Domain EntityId has no meaning, as these entities do not belong to a Domain
```

File: src/cpp/database/database.cpp [hunk 5]
```diff
@@ -777,6 +777,145 @@ const std::shared_ptr<const Entity> Database::get_entity(
...
+            break;
+        }
+        case EntityKind::INVALID:
+        {
+            throw BadParameter("INVALID EntityKind");
+        }
+        default:
+        {
+            throw Unsupported("Not implemented yet");
+        }
```
Comment (IkerLuengo) [inline src/cpp/database/database.cpp:905]: If we return `BadParameter` on both cases, we can combine these into `default`

**Review by IkerLuengo — CHANGES_REQUESTED** [review]

File: test/unittest/Database/DatabaseTests.cpp (hunk 3 above)
Comment (EduPonz) [reply test/unittest/Database/DatabaseTests.cpp:2120]: Done for the ones in which a second entity with the same name can be inserted

**Review by IkerLuengo — APPROVED** [review]
:+1:

**Pull request merged**